Update datatables

yubiuser opened the pull request:
**What does this PR aim to accomplish?:**

1) Updates `datatables`:

- `datatables` to `1.13.8`
- `datatables` plugin `button` to `2.4.2`
- ~~`datatables` plugin `select` to `1.7.0`~~ (was reverted during the development of this PR: the double check mark issue could not be solved)

2) Fixes a warning reported on the network table

> DataTables warning: DataTables warning: table id=network-entries - Requested unknown parameter 'ip' for row 0, column 1. For more information about this error, please see https://datatables.net/tn/4

`/network/device` won't return `ip` as top level key, but

```
      "macVendor": "Google, Inc.",
      "ips": [
        {
          "ip": "10.0.1.57",
          "name": null,
          "lastSeen": 1700344560,
          "nameUpdated": 1700344560
        }
```

Datatables data source was changed accordingly.

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

yubiuser committed:
Update datatables

Signed-off-by: Christian König <ckoenig@posteo.de>

yubiuser committed:
Fix datatable warning on network table

Signed-off-by: Christian König <ckoenig@posteo.de>

yubiuser commented:
Needs a bit of care
![Screenshot at 2023-11-18 23-14-40](https://github.com/pi-hole/web/assets/26622301/87392f93-0d4a-4f4a-9f0a-fadbecb68ab5)
![Screenshot at 2023-11-18 23-15-02](https://github.com/pi-hole/web/assets/26622301/3ea83444-37f1-4931-8852-62771835a402)

yubiuser committed:
Set  pagingTag: 'button'

Signed-off-by: Christian König <ckoenig@posteo.de>

rdwebdesign committed:
Add datatables bootstrap3 integration code (javascript)

- Add the main integration code for datatables
- Add the integration code for Buttons plugin

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Use the Bootstrap 3 version for buttons (CSS)

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Revert the style change for select plugin (back to 1.3.1)

This fixes the double checkbox icon used to select table rows

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Fix the selected row background and text colors

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign commented on scripts/pi-hole/js/groups-clients.js:
```diff
@@ -100,6 +100,7 @@ function initTable() {
       dataSrc: "clients",
       type: "GET",
     },
+    pagingTag: "button",
```
After adding the bootstrap3 integration, this seems to be unnecessary.

yubiuser replied:
I reverted the commit.

DL6ER commented:
Something isn't right with colors:

Unselected:
![Screenshot from 2023-11-19 07-30-37](https://github.com/pi-hole/web/assets/16748619/b32299a3-42c4-4009-8746-5cad650b3b4d)

Selected on `development-v6`:
![Screenshot from 2023-11-19 07-30-47](https://github.com/pi-hole/web/assets/16748619/488c66cc-947c-4e73-a7ab-b312113169a5)

---

So far, so good, but:

Selected on this branch (latest commit by @rdwebdesign):
![Screenshot from 2023-11-19 07-30-56](https://github.com/pi-hole/web/assets/16748619/542eeb21-1e36-48d5-b1d6-dc80b03c635e)

Selected on @yubiuser's last commit (https://github.com/pi-hole/web/pull/2852/commits/a8bb9ed201879864c6386ba2334d07ac543516ec):
![grafik](https://github.com/pi-hole/web/assets/16748619/e824c2d9-3a61-4c68-b483-740eb3611c6f)

rdwebdesign commented:
~~Hmmm... I forgot to add the CSS variables on the other themes.~~

EDIT:

Actually, the same is happening in all themes, but it's an issue in dark themes (the link color is dark).
This issue is related to some changes in datatables styles. 
Now they use a few CSS variables:
```css
:root {
	--dt-row-selected: 0, 136, 204;
	--dt-row-selected-text: 255, 255, 255;
	--dt-row-selected-link: 9, 10, 11;
	--dt-row-stripe: 0, 0, 0;
	--dt-row-hover: 0, 0, 0;
	--dt-column-ordering: 0, 0, 0;
	--dt-html-background: white;
}
```
But this is also changing the check mark to white, when the row is selected.

I will finish it later.

yubiuser committed:
Revert "Set  pagingTag: 'button'"

This reverts commit a8bb9ed201879864c6386ba2334d07ac543516ec.

Signed-off-by: Christian König <ckoenig@posteo.de>

rdwebdesign committed:
Use a modified version of datatables.min.css

- after the update, a few CSS rules were removed to keep the original look;
- a header containing a link to the removed code was added.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Rename the modified `datatables.min.css` to `datatables-pihole.min.css`

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser committed:
Revert datatables select to v 1.3.1

Signed-off-by: Christian König <ckoenig@posteo.de>

DL6ER reviewed (approved):
It seems to work everywhere I looked

sefinek commented:
![brave_BhJbAZBh3JLD](https://github.com/pi-hole/web/assets/92880265/ff2e7971-cc48-4bf8-aecc-e79a81f094b9)


https://github.com/pi-hole/web/assets/92880265/bbcbaa48-06ec-4bc3-9ae2-1494e24d9b8f

DL6ER commented:
Thanks for the report. We need to remove `<ul>` + `</ul>`

https://github.com/pi-hole/web/blob/4d3d64b90426fc97eb6dc0425b33ed8956f71c14/scripts/pi-hole/js/groups-domains.js#L444

and friends in similar files to fix this

Th3M3 commented:
Looks like the processing indicator also needs some patch to not just show a gray box.

![emptyanim](https://github.com/pi-hole/web/assets/28313514/2c318079-6b4b-471d-8bb9-5483660c373e)

rdwebdesign commented:
Ohh... I know what is the issue here. 

The updated datatables version introduced a new custom property: `--dt-row-selected`

This property accepts RGB colors _(actually, it just the numbers, not a real color code... this property looks strange)_, but I used a color in hex format.

yubiuser commented:
The updated datatables caused to much trouble and needed to many manual changes to your styles. This PR https://github.com/pi-hole/web/pull/2874 is reverting the changes. At one point we should update out web base (adminlte) and can re-apply updates to all components
